ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader.cpp

```diff
@@ -0,0 +1,78 @@
+#include "uploader.hpp"
+#include "libdatadog_helpers.hpp"
+
+using namespace Datadog;
+
+void
+DdogProfExporterDeleter::operator()(ddog_prof_Exporter* ptr) const
```

Comment:
where is this used?

Reply:
Blink and you'll miss it!  The `ddog_exporter` member of Uploader is a unique_ptr.  The template allows you to specify an optional destructor which will be called to cleanup the thing the pointer points to when it's done pointing.  That's what this does.